### PR TITLE
Set up release workflow

### PR DESCRIPTION
**Pull request by dbartholomae**

### CODE REVIEW

**Comment by kingthorin** [comment]
Was enforcement of conventional commits meant to help with release notes?

**Comment by dbartholomae** [comment]
Release notes and semantic versioning. The idea is that just based on the PR titles, the package can automatically deployed without needing to manually define a new version. This way, the release workflow can be fully automated.

**Comment by kingthorin** [comment]
I think I'm good with this one. It should be rebased now that the other PR is merged.

**Comment by dbartholomae** [comment]
For this to work, someone needs to set the secret `NPM_TOKEN` to an npm classic automation token with access to [@zaproxy/actions-common-scans](https://www.npmjs.com/package/@zaproxy/actions-common-scans).

**Comment by dbartholomae** [comment]
Yes, I would release every PR. I don't see any reason against it, and an important reason for it: If we don't then we need a manual step to decide when to release.

Good catch with dependabot: Normally, dependabot picks up on existing commit conventions, and names commits/PRs accordingly (see e.g. [this PR](https://github.com/dbartholomae/raw-data/pull/374)), but since the repo wasn't using conventional commits before, we need to explicitly tell dependabot (see latest commit).

**Comment by dbartholomae** [comment]
Not sure which part would be noisy - the only consumer of the package is the actions, which would get updates via dependabot on a schedule, so if we have 20 releases in an hour, they would still only get one PR to update. And dependabot is even smart enough to just update/replace a PR if the same package is available in a newer version.

**Comment by kingthorin** [comment]
Okay, I’m up for trying 🙂

**Comment by thc202** [comment]
>  And dependabot is even smart enough to just update/replace a PR if the same package is available in a newer version.

That still creates "noise" though, PR pushes create notifications. (Just saying.)

**Comment by dbartholomae** [comment]
Good point. I usually try to get to automatically merged dependabot PRs as soon as possible so that I don't even need to manually check those, but that might clash with the current approval policy. I would suggest to give it a try for now, as we don't have that many PRs in this repo, and to adjust course if it actually becomes a nuisance :)

**Review by kingthorin — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,3 +27,25 @@ jobs:
...
+          node-version: 16.x
+          cache: npm
+      - name: Install
+        run: npm ci
+      - name: Build
+        run: npm run build
+      - name: Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (dbartholomae) [inline .github/workflows/ci.yml:50]:
```suggestion
          NPM_TOKEN: ${{ secrets.ZAPBOT_NPM_TOKEN }}
```

**Comment by dbartholomae** [comment]
I've updated the workflow to use the new secret name

**Comment by thc202** [comment]
The latest commit needs to be signed-off.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**